File: backend/src/main/java/com/example/gestion_contribuable/DTO/FileDetailsDTO.java
package com.example.gestion_contribuable.DTO;

import lombok.Getter;
import lombok.Setter;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Getter
@Setter
public class FileDetailsDTO {
    private Long id;
    private String fileName;
    private String fileType;
    private byte[] fileData;
    private String ocrText;
    private Map<String,String> valeurs = new HashMap<>();

    public FileDetailsDTO(Long id, String fileName, String fileType, byte[] fileData, String ocrText, Map<String,String> valeurs) {
        this.id = id;
        this.fileName = fileName;
        this.fileType = fileType;
        this.fileData = fileData;
        this.ocrText = ocrText;
        this.valeurs = valeurs != null ? valeurs : new HashMap<>();
    }
}

File: frontend/src/app/models/FileDetailsDTO.ts
export interface FileDetailsDTO {
    id: number;
    fileName: string;
    fileType: string;
    fileData: string;
    ocrText?: string;
    valeurs?: { [key: string]: string};
}

export class FileDetailsDTO {
    constructor(
        public id: number,
        public fileName: string,
        public fileType: string,
        public fileData: string,
        public ocrText?: string,
        public valeurs?: { [key: string]: string}
    ) {}

    toBlob(): Blob {
        const byteCharacters = atob(this.fileData);
        const byteNumbers = new Array(byteCharacters.length);
        for (let i = 0; i < byteCharacters.length; i++) {
            byteNumbers[i] = byteCharacters.charCodeAt(i);
        }
        const byteArray = new Uint8Array(byteNumbers);
        return new Blob([byteArray], { type: this.fileType});
    }
}